src/components/SearchArea.tsx

```diff
@@ -0,0 +1,208 @@
+import algoliasearch from "algoliasearch/lite"
+import React, { PropsWithChildren, useEffect, useState } from "react"
+import Select from "react-select"
+
+import { Card } from "./Card"
+import { CardGroup } from "./CardGroup"
+
+import styles from "./SearchArea.module.css"
+import Grid from "./Grid"
+
+interface SearchAreaProps {
+  name: string
+  requirement?: "required" | "optional"
+  type: string
+}
+
+function ResultCardGroup(hits: any) {
+  return (
+    <CardGroup cols={3}>
+      {hits.map((hit: any) => {
+        return (
+          <Card
+            title={hit.moduleName}
+            key={hit.moduleName}
+            href={hit.docsRelativeUrl}
+            tags={[hit.mainRepoTitle]}
+            className={styles.card_container_max_width}
+          >
+            <div
+              className={styles.card_result}
+              dangerouslySetInnerHTML={{
+                __html: (hit.moduleDescription as string)
+              }}
+            ></div>
+          </Card>
+        )
+      })}
+    </CardGroup>
+  )
+}
+
+function NoResults() {
+  return (
+    <div className={styles.NoResultsContainer}>
+      <p className={styles.NoResults}>
+        No results found, please try another search
+      </p>
+    </div>
+  )
+}
+
+function CustomHits(hits: any[]) {
+  let newHits = hits["hits"]["searchHits"]
+
+  if (newHits.length === 0) {
+    return NoResults()
+  }
+
+  /*
+  Don't display search results where the module has been deprecated. We prefix the friendly name for these
+  modules with [DEPRECATED], so we filter for all hits where there is not a match for the prefix.
+  */
+  const activeHits = newHits.filter(
+    (hit: any) => !hit.moduleName.startsWith("[DEPRECATED]")
+  )
+
+  /*
+  Don't display modules where the description contains a note.
+  These modules all state they are not intended to be used directly, and the module description breaks out of the card
+  */
+  const onlyUseableModules = activeHits.filter(
+    (hit: any) => hit.moduleDescription.toLowerCase().indexOf("note") === -1
+  )
+
+  return onlyUseableModules.length > 0
+    ? ResultCardGroup(onlyUseableModules)
+    : NoResults()
+}
+
+const selectStyles = {
+  control: (base, state) => ({
+    ...base,
+    fontSize: "16px",
+    borderColor: state.isFocused ? "#6f5bd7" : base.borderColor,
+    boxShadow: state.isFocused ? "0 0 0 1px #6f5bd7" : base.borderColor,
+    "&:hover": {
+      borderColor: state.isFocused ? "#6f5bd7" : base.borderColor,
+      boxShadow: state.isFocused ? "0 0 0 1px #6f5bd7" : base.boarderColor,
+    },
+  }),
+}
+
+export const SearchArea: React.FunctionComponent<
+  PropsWithChildren<SearchAreaProps>
+> = ({ name, requirement, type, children }) => {
+  // TODO: Make using a configuration work here
```

Comment:
Did we want to leave this TODO in as a future reminder or pull out from this PR?

Reply:
im pulling it out now, i forgot to do it before i removed the WIP